aiohttp/web_app.py

```diff
@@ -171,6 +154,10 @@ def _set_loop(self, loop: Optional[asyncio.AbstractEventLoop]) -> None:
             stacklevel=2,
         )
 
+    @property
+    def state(self) -> _T:
```

Comment:
If I may ask, what is the thought process behind naming this `state`?

Reply:
Mainly that I saw the old (private) variable was named state. It makes as much sense to me as anything else.

If others come up with an agreed name, we can change it.

Reply:
My initial thought is that it might have an adverse affect of causing users to using it for reading and writing when it seems to me that the intended purpose is to act more as global context that is set up in the beginning and then only read from? I also don't know if that is the case though. This is also just a minor nit, I think the name `state` would probably be fine.

Reply:
While it's expected to have the top-level variables set on startup (before it freezes), you'll see plenty of examples of mutable objects being initialised for state, so I'm not sure that's an issue. For example, the web_ws.py example stores open web sockets in a list using the app object.

It's semi-read-only once frozen, so users trying to assign to the dict afterwards would get an error anyway, thus making it clear the intended behaviour.

Reply:
My bigger concern would be the verbosity of `request.app.state[...]`, so I'd also consider a shorter variable name, such as flask's `g`. Making it atleast a little bit shorter to write: `request.app.g[...]`.

Reply:
I hadn't thought about verbosity. What do you think about continuing to support `__getitem__`?

Reply:
I don't see any way to support static typing with it, and:
> There should be one-- and preferably only one --obvious way to do it.

Reply:
Fair enough, can't argue with zen.